src/Components/Components/src/RuntimeHelpers.cs

```diff
@@ -17,5 +20,23 @@ public static class RuntimeHelpers
         /// <param name="value"></param>
         /// <returns></returns>
         public static T TypeCheck<T>(T value) => value;
+
+        /// <summary>
+        /// Not intended for use by application code.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public static T TypeCheck<T>(Action value) => default(T);
```

Comment:
OK. Time for explanation - there's a problem here that we won't be able to fix in preview 3 (tooling only). This only concerns what happens in the editor, since that code is not getting updated.

Suppose you have the following component:

```html
@* MyButton.razor *@

<button onclick=@OnClick>Hi!</button>
@functions {
    [Parameter] EventCallback<UIMouseEventArgs> OnClick { get; set; }
}
```

and usage:
```html
<MyButton OnClick="@(() => { ... }) />
```

When the **OLD** compiler that runs in VS see this usage, it will generate code like:
```C#
builder.OpenComponent<MyButton>(0);
builder.AddAttribute(1, "OnClick", RuntimeHelpers.TypeCheck<EventCallback<UIMouseEventArgs>>(() => { ... }));
builder.CloseComponent();
```

The **OLD** compiler that runs in VS doesn't know about `EventCallback<>` and thus doesn't know how to generate a call to the `EventCallback.Factory.Create<>(...)` method, it does the old thing when you're editing. You might see a problem with this, because an `Action` is not an `EventCallback<UIMouseEventArgs>`.

Well, this is the reason why I added `TypeCheck<T>(Action value)` - it's a hack for preview 3 that will subvert the type check if you're using an `Action`. This will apply to a bunch of cases as well, so basically a bunch of code that wouldn't work at runtime will now compile. (I plan to update this to do a cast, so it will fail at runtime if needed).

I did a similar thing for `Func<Task>`. 

**now again** - I want to reiterate that this is only a problem for the editor, when means you get error squigglies. Everything works great when you build because the compiler is updated.

----

Now we get to the real problem. We would want to make a case like `Action<UIMouseEventArgs>` or `Func<UIMouseEventArgs, Task>` work as well. Well now we have a problem,

The compiler wrote:
```C#
RuntimeHelpers.TypeCheck<EventCallback<UIMouseEventArgs>>(() => { ... })
```

We want to define a method that accepts an `Action<U>` where `U` is substituted for `UIMouseEventArgs` in `EventCallback<UIMouseEventArgs>`.... and we're stuck.

This won't work because we can't change the method to `TypeCheck<T, U>`. C# doesn't allow you to partially specify type-arguments. We can't change what the compiler generates because that's in VS.

---

**TLDR;** So to reiterate, in the case you have a component that accepts an `EventCallback<T>` parameter and you want set that parameter to an `Action<T>` or `Func<T, Task>` this will cause a spurious error in the editor. 

Using `bind-...` on a component, or doing anything to an element has no problem because it doesn't do the same type checking.

Reply:
I'm going to try implicit conversion as well and see if that helps. I'm not sure you're allowed an implicit conversion from a delegate type.

Reply:
> So to reiterate, in the case you have a component that accepts an `EventCallback<T>` parameter and you want set that parameter to an `Action<T>` or `Func<T, Task>` this will cause a spurious error in the editor.

That's unfortunate. So will have to be a known issue for the preview 3 release, then?

Reply:
> That's unfortunate. So will have to be a known issue for the preview 3 release, then?

I don't see an alternative unless people have other suggestions. We can't change what the editor does without updating VS, and we can't update VS. We are limited to fixes that change the runtime. 

BTW the workaround for the user is to use `EventCallback.Factory.Create` for this case - or just ignore the error from the editor because everything works in the actual build.

Reply:
Understood. We can't update everything simultaneously. So I guess it's a known issue when we ship, but will go away when people update to VS 16 Preview 4?

Reply:
no, VS is done at this point. It has to go into 16.1

Reply:
After discussion of this, the plan is to add more special-casing for overloads of  `TypeCheck` so that the editing experience is nice.